iox_query_influxql/src/plan/planner.rs

```diff
@@ -1053,6 +1060,129 @@ impl<'a> InfluxQLToLogicalPlan<'a> {
+                        return Err(DataFusionError::NotImplemented(
+                            "database name in from clause".into(),
+                        ));
+                    }
+                    if qualified_name.retention_policy.is_some() {
+                        return Err(DataFusionError::NotImplemented(
+                            "retention policy in from clause".into(),
+                        ));
+                    }
+                    match &qualified_name.name {
+                        MeasurementName::Name(name) => {
+                            let name = name.as_str();
+                            if all_tables.contains(name) {
+                                out.insert(name);
+                            }
+                        }
+                        MeasurementName::Regex(regex) => {
+                            for name in &all_tables {
+                                let regex = parse_regex(regex)?;
+                                if regex.is_match(name) {
+                                    out.insert(name);
+                                }
+                            }
+                        }
+                    }
+                }
+
+                let mut out = out.into_iter().map(|s| s.to_owned()).collect::<Vec<_>>();
+                out.sort();
+                Ok(out)
+            }
+        }
+    }
+
+    fn show_field_keys_to_plan(
+        &self,
+        show_field_keys: ShowFieldKeysStatement,
+    ) -> Result<LogicalPlan> {
+        if show_field_keys.database.is_some() {
+            // How do we handle this? Do we need to perform cross-namespace queries here?
+            return Err(DataFusionError::NotImplemented(
+                "SHOW FIELD KEYS ON <database>".into(),
+            ));
+        }
+
+        let output_schema = Arc::new(ArrowSchema::new(vec![
+            ArrowField::new(INFLUXQL_MEASUREMENT_COLUMN_NAME, DataType::Utf8, false),
+            ArrowField::new("fieldKey", DataType::Utf8, false),
+            ArrowField::new("fieldType", DataType::Utf8, false),
+        ]));
+
+        let tables = self.expand_tables(show_field_keys.from)?;
+
+        let mut measurement_names_builder = StringBuilder::new();
+        let mut field_key_builder = StringBuilder::new();
+        let mut field_type_builder = StringBuilder::new();
+        for table in tables {
+            let Some(table_schema) = self.s.table_schema(&table) else {continue};
+            for (t, f) in table_schema.iter() {
+                let t = match t {
+                    InfluxColumnType::Field(t) => t,
+                    InfluxColumnType::Tag | InfluxColumnType::Timestamp => {
+                        continue;
+                    }
+                };
+                let t = match t {
+                    InfluxFieldType::Float => "float",
+                    InfluxFieldType::Integer => "integer",
+                    InfluxFieldType::UInteger => "uinteger",
```

Comment:
```suggestion
                    InfluxFieldType::UInteger => "unsigned",
```

Sourced from the list of InfluxQL data type names:

https://github.com/influxdata/influxql/blob/1ba470371ec093d57a726b143fe6ccbacf1b452b/ast.go#L177-L186